segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/mocks/InMemoryChunkStorageProvider.java

```diff
@@ -0,0 +1,289 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.mocks;
+
+import com.google.common.base.Preconditions;
+import io.pravega.segmentstore.storage.chunklayer.ChunkHandle;
+import io.pravega.segmentstore.storage.chunklayer.ChunkInfo;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * In-Memory mock for ChunkStorageProvider. Contents is destroyed when object is garbage collected.
+ */
+public class InMemoryChunkStorageProvider extends AbstractInMemoryChunkStorageProvider {
+    ConcurrentHashMap<String, InMemoryChunk> chunks = new ConcurrentHashMap<String, InMemoryChunk>();
```

Comment:
This is the last time I comment on this: 
1. Make all your fields private final
2. Do not expose maps or other internal data structures.